webapp/src/views/userSettings/accountSecurity/EnableMfaDialog.tsx

```diff
@@ -0,0 +1,172 @@
+  }, [user]);
+
+  const enableMfa = useApiMutation({
+    url: '/v2/user/mfa/totp',
+    method: 'put',
+    options: {
+      onSuccess: (r, v) => {
+        securityService.setToken(r.accessToken!);
+        message.success(<T keyName="account-security-mfa-enabled-success" />);
+        globalDispatch({ type: 'REFETCH_INITIAL_DATA' });
+        setRecoveryCodesPw(v.content['application/json'].password);
+      },
+    },
+  });
+
+  if (!user) return null;
+  if (recoveryCodesPw) {
+    return <MfaRecoveryCodesDialog password={recoveryCodesPw} />;
+  }
+
+  const encodedOtpName = encodeURIComponent(`Tolgee (${user.username})`);
+  const otpUri = `otpauth://totp/${encodedOtpName})?secret=${secret}`;
+
+  return (
+    <Dialog
+      open={true}
+      onClose={onDialogClose}
+      fullWidth
+      maxWidth="md"
+      data-cy="mfa-enable-dialog"
+    >
+      <DialogTitle data-cy="mfa-enable-dialog-title">
+        <T>account-security-mfa-enable-mfa</T>
+      </DialogTitle>
+      <DialogContent data-cy="mfa-enable-dialog-content">
+        <Box
+          display="flex"
+          flexDirection={{ xs: 'column', md: 'row' }}
+          gap={{ xs: 2, md: 8 }}
+        >
+          <Box flex={0} flexShrink={0}>
+            <Typography>
+              <T>account-security-mfa-enable-step-one</T>
+            </Typography>
+            <Box display="flex" justifyContent="center" flex={1}>
+              <WhiteBox p={2} mt={1} mb={1}>
+                <ReactQR value={otpUri} size={196} />
+              </WhiteBox>
+            </Box>
+            <Typography variant="body2" mb={0.5}>
+              <T>account-security-mfa-enable-manual-entry</T>
+            </Typography>
+            <code data-cy="mfa-enable-dialog-totp-key">
+              {secret.replace(/.{4}(?=.)/g, '$& ')}
+            </code>
+          </Box>
+          <Box flex={1}>
+            <Typography mb={2}>
+              <T>account-security-mfa-enable-step-two</T>
+            </Typography>
+            <StandardForm
+              onSubmit={(values) => {
+                enableMfa.mutate({
+                  content: { 'application/json': values },
+                });
+              }}
+              saveActionLoadable={enableMfa}
+              onCancel={() => onDialogClose()}
+              initialValues={
+                {
+                  totpKey: secret,
+                  password: '',
+                  otp: '',
+                } as TotpEnableDto
+              }
+              submitButtonInner={t('account-security-mfa-enable-mfa-button')}
+              validationSchema={Validation.USER_MFA_ENABLE}
+            >
+              <TextField
+                inputProps={{
+                  'data-cy': 'mfa-enable-dialog-otp-input',
+                }}
+                name="otp"
+                placeholder="000000"
+                label={t('account-security-mfa-otp-code')}
+              />
+              <TextField
```

Comment:
Could we maybe switch the order of the inputs? Since you need to be fast to type correct mfa to not to timeout, it's better to insert password first and mfa second.